### PR TITLE
Complete inclusion options

### LINKED ISSUE #21: Add inclusion options in Slicer GUI
Transferred from https://github.com/fepegar/EpilepsySemiology/issues/2, @thenineteen said:

need four cross-boxes in slicer under one heading **Ground Truth and Prior Inclusions** to filter results:

- [x] Seizure-Freedom (Engel Ia/Ib or ILAE 1 or 2*)
- [x] Concordance of Imaging and Neurophysiology
- [x] Cortical Stimulation
- [x] Patients from studies with pre-selected brain regions (Topology)

By default all the check boxes should be ticked. If any of the check boxes ARE UNTICKED then these scripts are run from [Semiology-Visualisation-Tool:](https://github.com/thenineteen/Semiology-Visualisation-Tool):

e.g. if the 2. concordance box is unticked

- [ ] Concordance

 the dataframe to use is under resources:

```
from SVT import resources

path_to_file = resources/DataFrames_Exclusions_df_postictalPEThyper_concordance_ET.pickle
with open(path_to_file, 'rb') as f:
    data = pickle.load(f)
 _, _, df_exclusions_concordance, _ = data
```

alternative solution for above:
after running the MEGA_ANALYSIS (or using the first pickled df):

```
from SVT.mega_analysis.crosstab.mega_analysis import exclusions

df_exclusions_concordance = exclusions(df, 
                POST_ictals=False,
                PET_hypermetabolism=False,
                SPECT_PET=False,
                CONCORDANCE=True)
```

The other most useful cross-out-box would be 

- [ ] ET: Topology/EZ

When unticked, it should run this from exclusions.py:
```
exclude_ET(df)
```

### PR DESCRIPTION
**Pull request by thenineteen**
added 3 exclusions, fixes #21 :

- paediatric <7 yrs
- spontaneous semiology
- postictals

@neurokleos there were some errors in the README (e.g. referring to seizure onset zone as resected area) which I've cleaned up. Please check you're happy with the updated README. If not, please either comment here (if you can check) otherwise make a new issue after this is merged?

I've also updated the  tooltips to reflect exactly what is being excluded for our sanity. 

I would be grateful if @fepegar could clarify exacrtly what functions are run when each of the checkboxes are ticked/unticked ground truth and publication approaches tab.

### CODE REVIEW

**Comment by thenineteen** [comment]
@fepegar 

maybe you can add the slicer inclusion options to this pull request and then squash and merge all at the same time?

**Comment by thenineteen** [comment]
I'm leaving one test deliberately commented out, as will need it again in future when we upload the updated excel

File: slicer/SemiologyVisualization.py [hunk 1]
```diff
@@ -161,10 +161,11 @@ def getInclusionsWidget(self):
...
     self.seizureSemiologyCheckBox.setToolTip(
-      'When the paper selects a sample of patients based on their siezure semiology'
-      ' (e.g. genital automatisms, nose-wiping, gelastic, ictal kissing),'
-      ' or other factors (specific techniques used, specific associated conditions'
-      ' like FCD etc.), and provides details of epileptogenic zone localisation/lateralisation'
+      'When the paper pre-selects a sample of patients based on their seizure semiology'
+      ' (e.g. nose-wiping, gelastic, ictal kissing),'
+      ' or takes a cohort of unselected patients with epilepsy,'
+      ' or pre-selects based on other non-topological factors (specific techniques or conditions'
+      ' e.g. FCD), and provides details of epileptogenic zone localisation/lateralisation'
```
Comment (fepegar) [inline slicer/SemiologyVisualization.py]: This is pasted from the README, so you might want to change the text there as well.

File: slicer/SemiologyVisualization.py [hunk 2]
```diff
@@ -38,9 +38,9 @@ def __init__(self, parent):
     self.parent.dependencies = []
     self.parent.contributors = [
       "Fernando Perez-Garcia (3D Slicer Module and data processing)",
-      "Ali Alim-Marvasti (data collection and processing)",
+      "Ali Alim-Marvasti (data collection and processing modules)",
```
Comment (fepegar) [inline slicer/SemiologyVisualization.py:41]: Can you change it for me as well, then?

File: slicer/SemiologyVisualization.py [hunk 3]
```diff
@@ -131,7 +131,7 @@ def getInclusionsWidget(self):
       'Invasive EEG recording and/or electrical stimulation, mapping seizure semiology'
     )
     self.concordanceCheckBox.setToolTip(
-      'Multimodal concordance between brain imaging and non-invasive neurophysiology'
+      'Multimodal concordance between brain imaging and neurophysiological'
```
Comment (fepegar) [inline slicer/SemiologyVisualization.py:138]: Why not just "neurophysiology"?

File: slicer/SemiologyVisualization.py [hunk 4]
```diff
@@ -156,19 +162,25 @@ def getInclusionsWidget(self):
     self.seizureSemiologyCheckBox.setEnabled(False)  # not implemented yet
 
     self.epilepsyTopologyCheckBox.setToolTip(
-      'When the paper selects a sample of patients based on their established'
-      ' epileptogenic zone and/or site of surgical resection (seizure onset zone),'
-      ' and describes the related seizure semiology, e.g. papers looking at TLE, FLE, OLE, etc.'
+      'When the paper pre-selects samples of patients based on their established epileptogenic zone'
+      '(site of surgical resection) or seizure onset zone (neurophysiological/anatomical), and'
+      'describes the related seizure semiology - e.g. articles looking at TLE, FLE, OLE.'
+      ''
```
Comment (fepegar) [inline slicer/SemiologyVisualization.py:168]: This line doesn't do anything. You can use `\n`, but using two paragraphs in a tooltip seems excessive.

File: slicer/SemiologyVisualization.py [hunk 5]
```diff
@@ -156,19 +162,25 @@ def getInclusionsWidget(self):
...
+      'When unticked, ALL epilepsy topology data are excluded, EVEN if there are other approaches.'
```
Comment (fepegar) [inline slicer/SemiologyVisualization.py:169]: I think we can stick to lowercase.

File: slicer/SemiologyVisualization.py (hunk 1 above)
Comment (thenineteen) [reply slicer/SemiologyVisualization.py]: yes readme is updated on this too

File: slicer/SemiologyVisualization.py (hunk 2 above)
Comment (thenineteen) [reply slicer/SemiologyVisualization.py]: have a look now

File: slicer/SemiologyVisualization.py (hunk 3 above)
Comment (thenineteen) [reply slicer/SemiologyVisualization.py:138]: because it had findings. edited as to your preference!

File: slicer/SemiologyVisualization.py (hunk 4 above)
Comment (thenineteen) [reply slicer/SemiologyVisualization.py:168]: I think it's important for the user as some exclude all the cases, others only if it is the only critera.

File: slicer/SemiologyVisualization.py (hunk 5 above)
Comment (thenineteen) [reply slicer/SemiologyVisualization.py:169]: It's for emphasis, as above. Please leave for now

File: slicer/SemiologyVisualization.py [hunk 6]
```diff
@@ -37,10 +37,10 @@ def __init__(self, parent):
     self.parent.categories = ["Epilepsy Semiology"]
     self.parent.dependencies = []
     self.parent.contributors = [
-      "Fernando Perez-Garcia (3D Slicer Module and data processing)",
-      "Ali Alim-Marvasti (data collection and processing modules)",
-      "Gloria Romagnoli (data collection)",
-      "John Duncan (supervisor)",
+      "Fernando Perez-Garcia (SVT software: Design, 3D Slicer Module, Tests, and GUI API)",
```
Comment (fepegar) [inline slicer/SemiologyVisualization.py:40]: I don't think it's fair to say that all I've done related to `mega_analysis`/"MegaAnalysis Preprocessing Modules" is "Tests"...

File: slicer/SemiologyVisualization.py (hunk 4 above)
Comment (fepegar) [reply slicer/SemiologyVisualization.py:168]: I just meant that line 168 doesn't change the tooltip, I didn't mean that the text was useless. Maybe it wasn't clear, I don't know why 4 lines show up, sorry!

File: slicer/SemiologyVisualization.py (hunk 6 above)
Comment (fepegar) [reply slicer/SemiologyVisualization.py:40]: Another option, maybe easier, is just to not specify what each of us has done.

**Pull request merged**